### PR TITLE
feat: include the wallet-id responsible for the log

### PR DESCRIPTION
**Pull request by r4mmer**
### Acceptance Criteria
- Include the wallet-id on the logs where possible

### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -39,6 +39,18 @@
...
+      "rules": {
+        "no-console": "error"
+      }
+    },
+    {
+      "files": [ "src/plugins/**" ],
+      "rules": {
+        "no-console": "off"
+      }
+    },
```
Comment (r4mmer) [inline .eslintrc:53]: These 2 entries make the rule `no-console` an error on all source files except the plugins.
The console should be used on the scripts, tests and plugins directly but we are changing to use only from the logger when inside the wallet routes.

I prefer setting `no-console` as off by default so any new directory is not affected by this rule.

File: src/app.js [hunk 2]
```diff
@@ -36,7 +38,7 @@ const createApp = config => {
   app.use(ConfigErrorHandler);
   app.use(ReadonlyErrorHandler);
   app.use((err, req, res, next) => {
-    console.error(err.stack);
+    req.logger.error(err.stack);
```
Comment (tuliomir) [inline src/app.js:41]: thought: The patch coverage for this PR is low and is not passing the CI. I believe one of the reasons is that it that it targets error handling codes from other PRs that were never covered: the scope of this PR overlaps with the technical debt of many others.

Probably it will be necessary to add tests increasing the coverage over the updated lines, increasing the original scope of this PR.

File: src/services/logger.service.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  } else {
+    // There is no wallet for this request
+    // will return the app logger instance.
+    return buildAppLogger();
+  }
+
+  const logger = walletLoggers.get(walletId);
+  if (!logger) {
+    // Uninitialized wallet, should use the app logger.
+    return buildAppLogger();
```
Comment (pedroferreira1) [inline src/services/logger.service.js:47]: Can we have here the app logger with the walletId parameter? Maybe we could have a log with `non-initialized wallet {wallet-id}` or something like that in this case. What do you think?

File: src/logger.js [hunk 4]
```diff
@@ -57,33 +63,61 @@ function buildLogger(config) {
...
-
-  return logger;
+  return appLogger;
 }
 
 // This fixes some logs where Github code scanning complains about a `Log Injection` possibility
-const sanitizeLogInput = input => String(input).replace(/\n|\r/g, '');
+function sanitizeLogInput(input) {
+  return String(input).replace(/\n|\r/g, '');
+}
```
Comment (pedroferreira1) [inline src/logger.js:110]: Why did you change this?

File: src/logger.js [hunk 5]
```diff
@@ -7,14 +7,20 @@
 
 import util from 'util';
 import winston from 'winston';
+import { getConfig } from './settings';
+/** @import { Configuration } from './settings' */
```
Comment (pedroferreira1) [inline src/logger.js:11]: Should we remove this?

File: src/services/fireblocks.service.js [hunk 6]
```diff
@@ -253,11 +253,13 @@ class FireblocksClient {
         && e.response.data
         && e.response.data.code === TX_ALREADY_EXISTS_ERROR_CODE
       )) {
+        // eslint-disable-next-line no-console
```
[the remark this answers is not in the document]
Comment (pedroferreira1) [reply src/services/fireblocks.service.js:256]: It would be good, if possible

File: src/helpers/tx.helper.js [hunk 7]
```diff
@@ -226,10 +229,10 @@ async function getTx(wallet, id) {
       return response.tx;
     }
 
-    console.warn('Failed to get transaction from fullnode.', response.message);
+    logger.warn('Failed to get transaction from fullnode.', response.message);
```
[the remark this answers is not in the document]
Comment (pedroferreira1) [reply src/helpers/tx.helper.js:232]: The `AppLogger` as default is good

File: src/controllers/fireblocks/fireblocks.controller.js [hunk 8]
```diff
@@ -63,7 +66,7 @@ async function startFireblocksWallet(req, res) {
   if (!(config.fireblocksUrl
         && config.fireblocksApiKey
         && (config.fireblocksApiSecret || config.fireblocksApiSecretFile))) {
-    console.error('Error starting wallet because fireblocks is not configured.');
+    logger('Error starting wallet because fireblocks is not configured.');
```
Comment (pedroferreira1) [inline src/controllers/fireblocks/fireblocks.controller.js:69]:
```suggestion
    logger.error('Error starting wallet because fireblocks is not configured.');
```

File: src/helpers/tx.helper.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (r4mmer) [reply src/helpers/tx.helper.js:232]: >If we forget the optional parameter options.logger this line will crash, right?

But `options.logger` is not optional, the idea is that it always should be passed.

File: src/services/fireblocks.service.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (r4mmer) [reply src/services/fireblocks.service.js]: Since the wallet can only start 1 fireblocks wallet per instance I thought that the fireblocks specific logs would not create a doubt on which wallet it comes from.
But I guess it would be nicer to be more consistent with this logger, so I changed to include the logger instance

File: src/logger.js (hunk 4 above)
Comment (r4mmer) [reply src/logger.js:110]: We were having errors on the tests because `sanitizeLogInput` was not a function.
I believe that the issue was on how the exports were being organized, since we added more exports in the file it changed the behavior.

The method is the same, only that now it is a function.

File: src/logger.js (hunk 5 above)
Comment (r4mmer) [reply src/logger.js:11]: This is not a comment, it is a JSDoc import so we can use the `Configuration` defined as a `@typedef` in `src/settings.js`.

We can remove it, personally I just like having the argument types since it helps with development (LSP completion and checks) and it helps with documentation since it helps understand what is expected to be in the config object.

File: src/index.js [hunk 9]
```diff
@@ -28,10 +28,12 @@ async function startHeadless() {
       process.argv.slice(2),
       { silent: true, env: process.env },
     );
+    // eslint-disable-next-line no-console
```
[the remark this answers is not in the document]
Comment (r4mmer) [reply src/index.js]: Done.
The application logger has not yet been initialized.
Maybe later we can change the logger initialization so all `console.log`s are from the logger.

File: src/app.js (hunk 2 above)
Comment (r4mmer) [reply src/app.js:41]: The reason was that some tests were failing.
Once the issue was resolved the patch coverage came up to 87%

File: src/services/logger.service.js (hunk 3 above)
Comment (r4mmer) [reply src/services/logger.service.js]: Done

**Review by pedroferreira1 — APPROVED** [review]

File: src/helpers/tx.helper.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (tuliomir) [reply src/helpers/tx.helper.js:232]: It seems the only place this is used is inside the `decodeTx` controller handler. In this case, I think it's not worth having too much effort handling this problem.

But I think letting the javascript throw an object destructuring error is a sub-optimal approach to a method meant to serve as a helper.

**Review by tuliomir — APPROVED** [review]
